### PR TITLE
Add payload bandwidth support on switchtec-user command gui and bw. 

### PR DESCRIPTION
**Pull request by JoeyZhang-Microsemi**
Switchtec firmware will support payload bandwidth for performance monitor. We add an option --bw_type for switchtec-user command gui and bw to support the payload bandwidth.

### CODE REVIEW

File: lib/pmon.c [hunk 1]
```diff
@@ -348,6 +348,60 @@ void switchtec_bwcntr_sub(struct switchtec_bwcntr_res *new,
 	new->ingress.comp -= old->ingress.comp;
 }
 
+/**
+ * @brief Set bandwidth type for a number of ports
+ * @param[in]  dev		Switchtec device handle
+ * @param[in]  nr_ports		Number of ports to set
+ * @param[in]  phys_port_ids	The physical ids for each port to set
+ * @param[in]  bw_type		0 if raw bandwidth, 1 if payload bandwidth
```
Comment (lsgunth) [inline lib/pmon.c]: Please use an enum for bandwidth type so the code will be more descriptive than 0 and 1.

File: cli/main.c [hunk 2]
```diff
@@ -148,12 +150,14 @@ static int gui(int argc, char **argv)
 		 "gui refresh period in seconds (default: 1 second)"},
 		{"duration", 'd', "", CFG_INT, &cfg.duration, required_argument,
 		 "gui duration in seconds (-1 forever)"},
+		{"bw_type", 'b', "NUM", CFG_POSITIVE, &cfg.bw_type,
+		 required_argument, "gui bw type: 0 raw(default), 1 payload"},
```
Comment (lsgunth) [inline cli/main.c]: Please use a CFG_CHOICES for the options or make a different option for each type (--payload, --raw). 0 and 1 are completely unintuitive.

**Comment by JoeyZhang-Microsemi** [comment]
Hi Logan,

Thanks for your review and comments. I has updated the code.

Thanks
Joey

File: cli/main.c [hunk 3]
```diff
@@ -275,20 +285,37 @@ static int bw(int argc, char **argv)
...
 		.meas_time = 5,
+		.bw_type = SWITCHTEC_BW_TYPE_RAW,
 	};
+
+	const struct argconfig_choice types[] = {
+		{"RAW", SWITCHTEC_BW_TYPE_RAW, "Get the raw bandwidth"},
+		{"PAYLOAD", SWITCHTEC_BW_TYPE_PAYLOAD, "Get the payload bandwidth"},
+		{}
+	};
+
```
Comment (lsgunth) [inline cli/main.c]: Don't repeat yourself. Make this 'static const', file scope and a more descriptive name.

Comment (JoeyZhang-Microsemi) [reply cli/main.c]: Thank you very much! I updated my codes.

**Pull request merged**